index.js

```diff
@@ -69,7 +69,8 @@ function _createXHR(options) {
         if (xhr.response) {
             body = xhr.response
         } else if (xhr.responseType === "text" || !xhr.responseType) {
-            body = xhr.responseText || xhr.responseXML
+            var xml = (xhr.status === 204 && xhr.responseXML && xhr.responseXML.documentElement.nodeName === "parsererror") ? null : xhr.responseXML
```

Comment:
The more I look at it, the more I like the idea. It really does just eliminate the firefox bug and change no existing logic.